oap-server/analyzer/meter-analyzer/src/main/java/org/apache/skywalking/oap/meter/analyzer/prometheus/rule/Rules.java

```diff
@@ -40,36 +49,71 @@
 public class Rules {
     private static final Logger LOG = LoggerFactory.getLogger(Rule.class);
 
-    public static List<Rule> loadRules(final String path) throws ModuleStartException {
+    public static List<Rule> loadRules(final String path) throws IOException {
         return loadRules(path, Collections.emptyList());
     }
 
-    public static List<Rule> loadRules(final String path, List<String> enabledRules) throws ModuleStartException {
-        File[] rules;
-        try {
-            rules = ResourceUtils.getPathFiles(path);
-        } catch (FileNotFoundException e) {
-            throw new ModuleStartException("Load fetcher rules failed", e);
-        }
-        return Arrays.stream(rules)
-            .filter(File::isFile)
-            .map(f -> {
-                try (Reader r = new FileReader(f)) {
-                    String fileName = f.getName();
-                    int dotIndex = fileName.lastIndexOf('.');
-                    fileName = (dotIndex == -1) ? fileName : fileName.substring(0, dotIndex);
-                    if (!enabledRules.contains(fileName)) {
-                        return null;
+    public static List<Rule> loadRules(final String path, List<String> enabledRules) throws IOException {
+
+        final Path root = ResourceUtils.getPath(path);
+
+        Map<String, Boolean> formedEnabledRules = enabledRules
+                .stream()
+                .map(rule -> {
+                    rule = rule.trim();
+                    if (rule.startsWith("/")) {
+                        rule = rule.substring(1);
+                    }
+                    if (!rule.endsWith(".yaml") && !rule.endsWith(".yml") && !rule.endsWith("*")) {
```

Comment:
We can use this to ensure that all listed files are `.yaml` or `.yml`

```suggestion
                    if (!rule.endsWith(".yaml") && !rule.endsWith(".yml")) {
```

So (2)